src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -83,20 +93,52 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             // Description
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
```

Comment:
Please do not refactor this method. @rynowak did things this way to make it easy to see everything about a property in one place and performance doesn't outweigh that readability by a long shot. Note `ModelMetadata` details are cached and these providers run very few times.